### DISCUSSION TITLE
secp256k1-zkp pedersen commitments, borromean ring signatures, zk range proofs and surjection proofs

### DISCUSSION THREAD

**Discussion by asoltys** [paulmillr/noble-curves#147] [General]
Hi Paul,

https://github.com/BlockstreamResearch/secp256k1-zkp contains functions that are used on the Bitcoin Liquid sidechain for confidential transactions.

I thought it might be fun to try porting them to JS. I started with the rangeproof sign function:

https://github.com/paulmillr/noble-curves/compare/ad263ed2957b21a6025e6843f0e827f2a31f0bc2...asoltys:noble-curves:a08625cc90750a1c67467ed56cc7d6c4206d51da

I'm a little out of my element here and have been leaning on ChatGPT a lot. 

I figured I'd open an issue before going too much further to see if this might be something you'd entertain adding to this repo and reviewing at some point or if you think there's a better place for it.

By the way https://github.com/vulpemventures/secp256k1-zkp creates WASM bindings so that the C lib can be used from JavaScript by https://github.com/vulpemventures/liquidjs-lib  

I have ambitions to work on a scure-liquid-signer lib analogous to your scure-btc-signer

**Comment by paulmillr** [comment]
Sounds great! I have no plans in writing the library myself (for now). I will add your library to documentation / resources once finished, of course.

How's liquid adoption these days?

**Comment by asoltys** [comment]
Sounds good, thanks! Liquid adoption is still pretty low but I still have a soft spot for it and look to it as a kind of middle ground between self-custodial base layer BTC and fully custodial Lightning in terms of wallet UX.

**Comment by Giszmo** [comment]
Any progress on borromean ring signatures?

**Comment by paulmillr** [comment]
* MuSig2 is now available here https://github.com/paulmillr/scure-btc-signer
* zk proof stuff would reside here https://github.com/paulmillr/micro-zk-proofs
